### ISSUE TITLE
`--annotation` option does not update the annotations.

### ISSUE THREAD

**Issue by HolyGuacamole** [thtrieu/darkflow#121]
If you specify the loctation of the annotations using `--annotation` the annotations are parsed and saved.
When you want to train any other (or the same) network with the same labels, the same set of annotations is loaded even if the `--annotation`  option points to a different set of annotations.

I deleted the file `net/yolo/parse-history.txt` to get the annotations updated.

**Comment by thtrieu** [comment]
Hmmm, that is one interesting bug that requires rethinking of this repo's design, any suggestion?

**Comment by HolyGuacamole** [comment]
Always parse the annotations instead of saving and loading them?

Another possibility is to make the parsing of the annotations a separate step that should be done before training.
The file with the parsed annotation can then be given with the `--annotation` option.

**Comment by thtrieu** [comment]
Yeah I'm thinking about 2nd possibility. Will do it soon.

**Comment by mastersplinter** [comment]
There is another issue to with parsing the history.

If there is a failure during the initial parsing of the annotations, e.g. because they're in the wrong format, the `.parsed` file will be be partially written. When the user re-runs `.flow` it will error because the file is incomplete.

> **HolyGuacamole** 
> Always parse the annotations instead of saving and loading them?

This can be achieved by removing the the history sections of the `parse` function in `data.py`.

@thtrieu I've raised #194 which removes the saving of history for the parse function.

I went with this option as opposed to option2:
> parsing of the annotations a separate step
because I wasn't sure if it would be better to:
- update `parse` to check whether `--annotations` is a directory or a `.parsed` file
- or if it would be better to make a new tag, e.g. `--annotation-history`

**Comment by HolyGuacamole** [comment]
I think the option you chose, gives the best solution for now.
